api/account/api.go

```diff
@@ -181,23 +181,39 @@ func convertToAccreditations(ctx context.Context, values []string, logger keyclo
 	return &accreds
 }
 
-// ConvertToKCUser creates a KC user representation from an API user
-func ConvertToKCUser(user UpdatableAccountRepresentation) kc.UserRepresentation {
-	var userRep kc.UserRepresentation
+func defaultStringPtr(value *string, defaultValue *string) *string {
+	if value != nil {
+		return value
+	}
+	return defaultValue
+}
 
-	userRep.Username = user.Username
-	userRep.Email = user.Email
-	userRep.FirstName = user.FirstName
-	userRep.LastName = user.LastName
+// MergeUser merge new values into existing KC user representation
+func MergeUser(userRep *kc.UserRepresentation, user UpdatableAccountRepresentation) {
+	userRep.Username = defaultStringPtr(user.Username, userRep.Username)
+	// Don't set email as email and phoneNumber have specific validation rules
+	//userRep.Email = defaultStringPtr(user.Email, userRep.Email)
+	userRep.FirstName = defaultStringPtr(user.FirstName, userRep.FirstName)
+	userRep.LastName = defaultStringPtr(user.LastName, userRep.LastName)
 
 	var attributes = make(kc.Attributes)
```

Comment:
Why we don't use the duplicateAttribute method ?

Reply:
Because it does not exist... and we are merging values, not duplicating them